libs/common/src/blockchain/extrinsic.ts

```diff
@@ -63,19 +63,17 @@ export class Extrinsic<T extends ISubmittableResult = ISubmittableResult, C exte
   }
 
   public signAndSend(nonce?: number): Promise<[Hash, EventMap]> {
-    return firstValueFrom(this.extrinsic.signAndSend(this.keys, { nonce }).pipe(filter(({ status }) => status.isInBlock || status.isFinalized))).then(
-      ({ status, events, txHash }) => {
-        if (status.isInBlock || status.isFinalized) {
-          const eventMap: EventMap = {};
-          events.forEach((record: EventRecord) => {
-            const { event } = record;
-            eventMap[eventKey(event)] = event;
-          });
-          return [txHash, eventMap];
-        }
-        throw new Error(`Transaction failed to finalize: ${txHash}`);
-      },
-    );
+    return firstValueFrom(this.extrinsic.signAndSend(this.keys, { nonce })).then(({ status, events, txHash }) => {
+      if (status.isFinalized || status.isInBlock) {
```

Comment:
It's been a while that I've checked different statuses but we are treating `inblock` and `finalized` as the same. What would happen if in block but not finalized later? Not sure how likely it is or should we even care about it

Reply:
I dont think it will ever be true in a realistic sense , not sure in instant sealing it  is true and is truly instant